api/src/test/java/org/jboss/cdi/api/test/annotated/AnnotatedTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.cdi.api.test.annotated;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import javax.enterprise.inject.spi.Annotated;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+/**
+ * @author John D. Ament
+ */
+public class AnnotatedTest {
```

Comment:
I'm inclined to believe that the correct solution is to:

- Leave `Annotated` with just the method declaration
- Add default implementations to `AnnotatedMember`, `AnnotatedParameter` and `AnnotatedType`.  The reason being is that the target of what is being scanned for annotations differs in each of those.  So the default impl in the root `Annotated` doesn't make much sense since there's no real target.

And then add tests for those three.

There's still an underlying problem, and this is where we're getting some discrepancies.  The current API is focused on working in a compatible way - however we have other areas that are not compatible.  So either `getAnnotations` and all related methods (`isAnnotationPresent` for instance) either should behave like the JDK or not like the JDK and that should be consistent in this class.  The current impl of `getAnnotations` is not consistent with the JDK.